src/main.cpp

```diff
@@ -2661,19 +2661,23 @@ void txRxProcessing()
                     g_recFromModulatorSamples -= nfreedv;
                 }
             }
-
-            // Attenuate signal prior to output
-            for (int i = 0; i < nfreedv; i++)
-            {
-                outfreedv[i] = outfreedv[i] * ((double)g_txLevel/100);
-            }
             
             // output one frame of modem signal
 
             if (g_analog)
                 nout = resample(cbData->outsrc1, outsound_card, outfreedv, g_soundCard1SampleRate, freedvInterface.getTxSpeechSampleRate(), 10*N48, nfreedv);
             else
                 nout = resample(cbData->outsrc1, outsound_card, outfreedv, g_soundCard1SampleRate, freedv_samplerate, 10*N48, nfreedv);
+            
+            // Attenuate signal prior to output
+            double dbLoss = g_txLevel / 10; // (-10) + (0 - (-10)) * ((double)g_txLevel)/100.0;
+            double scaleFactor = exp(dbLoss/20.0 * log(10.0));
+            
```

Comment:
I fprintf-ed `scaleFactor` and it check out dB value == linear :+1:

Reply:
The line you commented on didn't look exactly right so I had to make sure. Looks like 10.0 instead of 10 is used as expected (to avoid casting issues).